src/components/views/dialogs/AddressPickerDialog.js

```diff
@@ -502,28 +505,38 @@ module.exports = React.createClass({
         };
 
         // wait a bit to let the user finish typing
-        return Promise.delay(500).then(() => {
-            if (cancelled) return null;
-            return MatrixClientPeg.get().lookupThreePid(medium, address);
-        }).then((res) => {
-            if (res === null || !res.mxid) return null;
-            if (cancelled) return null;
+        await Promise.delay(500);
+        if (cancelled) return null;
 
-            return MatrixClientPeg.get().getProfileInfo(res.mxid);
-        }).then((res) => {
-            if (res === null) return null;
+        try {
+            const authClient = new IdentityAuthClient();
+            const isAccessToken = await authClient.getAccessToken();
             if (cancelled) return null;
+
+            const lookup = await MatrixClientPeg.get().lookupThreePid(
+                medium,
+                address,
+                undefined /* callback */,
+                isAccessToken,
+            );
+            if (cancelled || lookup === null || !lookup.mxid) return null;
+
+            const profile = await MatrixClientPeg.get().getProfileInfo(lookup.mxid);
+            if (cancelled || profile === null) return null;
+
             this.setState({
                 suggestedList: [{
                     // a UserAddressType
                     addressType: medium,
                     address: address,
-                    displayName: res.displayname,
-                    avatarMxc: res.avatar_url,
+                    displayName: profile.displayname,
+                    avatarMxc: profile.avatar_url,
                     isKnown: true,
                 }],
             });
-        });
+        } catch (e) {
+            console.error(e);
```

Comment:
do we not want to report this to the user?

Reply:
Ah, I was just maintaining previous functionality, but it's probably good to show something... Many of the errors I've seen while testing are generic CORS errors, so I'm not quite sure what the best message would be. I've changed to at least say something's wrong.

Reply:
I can believe it's mostly CORS errors :(

ops might want to hear about it btw, given it's likely a reverse proxy configuration issue.

as for the wording: saying something is better than nothing, so thank you!

Reply:
I filed a [Sydent issue](https://github.com/matrix-org/sydent/issues/183) to add CORS headers to endpoint that 404, which would have helped for most of it. I guess that's also unfixed for Synapse, so I won't hold my breath...

(Maybe we do want a general CORS change on the reverse proxy for most apps, if that's what you meant...?)

Reply:
Ops has been trying to track down all the places where CORS headers aren't sent. I believe they need explicit endpoints that cause problems so they can fix it for matrix.org

It's generally considered a misconfiguration of the service provider to not provide CORS headers on errors/unknown endpoints, so Riot doesn't do much about it.

Reply:
Right, thanks for the context. I added it to the Ops task list as well.